app.rb

```diff
@@ -0,0 +1,88 @@
+require './book'
+require './person'
+require './rental'
+require './classroom'
+require './teacher'
+require './student'
+
+# List all books
+def list_all_books(books)
+  books.each do |book|
+    puts "Title: #{book.title}, Author: #{book.author}"
+  end
+end
+
+# List all people
+def list_all_people(people)
+  people.each do |person|
+    puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+  end
+end
+
+# Create a person (teacher or student)
+def create_person
+  print 'Enter name: '
+  name = gets.chomp
+  print 'Enter age: '
+  age = gets.chomp.to_i
+  print 'Enter type (teacher/student): '
+  type = gets.chomp.downcase
+
+  if type == 'teacher'
+    print 'Enter specialization: '
+    specialization = gets.chomp
+    person = Teacher.new(age, specialization, name: name)
+  elsif type == 'student'
+    person = Student.new(age, name: name)
+  else
+    puts 'Invalid person type.'
+    return nil
+  end
+
+  person
+end
+
+# Create a book
+def create_book(books)
+  print 'Enter title: '
+  title = gets.chomp
+  print 'Enter author: '
+  author = gets.chomp
+
+  book = Book.new(title, author)
+  books << book
+  book
+end
+
+# Create a rental
+def create_rental(books, people)
+  print 'Enter date (YYYY-MM-DD): '
+  date = gets.chomp
+  print 'Enter book title: '
+  title = gets.chomp
+  print 'Enter person id: '
+  person_id = gets.chomp.to_i
+
+  book = books.find { |b| b.title == title }
+  person = people.find { |p| p.id == person_id }
+
+  if book && person
+    Rental.new(date, book, person)
+  else
+    puts "Book with title '#{title}' or person with id '#{person_id}' not found."
+  end
+end
+
+# List all rentals for a given person id
+def list_rentals_for_person(person_id, people)
+  person = Person.find_by_id(person_id, people)
+
+  if person.nil?
+    puts "Person with ID #{person_id} not found."
+    return
+  end
+
+  person.rentals.each do |rental|
+    puts "Date: #{rental.date}, Book: #{rental.book.title}"
+  end
+end
```

Comment:
Great job so far. Most of the commands except enter 5 and 6 are not working properly. you can see the error below the screenshots so I can't see your UI whether it's accepting the correct command or not. 
For example, the problem for the `create person` method was expecting 0 arguments, but you were passing it 2 arguments: name and age. This is because you were calling the create_person method like this:
create_person(name, age)
To fix the error, you should have removed the arguments from the call to the create_person method in the main loop:

`def create_person`
  `name = gets.chomp`
  `age = gets.chomp.to_i`
 `type = gets.chomp.downcase`

  `if type == 'teacher'`
    `print 'Enter specialization:`
    `specialization = gets.chomp`
    `Teacher.new(age, specialization, name: name)`
  `elsif type == 'student `
  `Student.new(age, name: name)`
  `else`
    `puts 'Invalid person type.'
    `Teacher.new or Student.new`
  `end`
`end`

Kindly fix all the other commands like this./
![image](https://github.com/Omwomo/school-library/assets/101523582/582e8570-dc97-4076-8ab1-273f662f75be)

Reply:
Hello, I tried fixing the issue, for example in the create_person as you are saying, in my code is takes no argument.  "def create_person
  print 'Enter name: '
  name = gets.chomp
  print 'Enter age: '
  age = gets.chomp.to_i
  print 'Enter type (teacher/student): '
  type = gets.chomp.downcase". Yet I'm still getting the error of wrong number of arguments, given 2, expected 0.
Can you help understand how I am calling the method this way 'create_person(name, age)' as you have mentioned in the comment